authz/audit/audit_logger.go

```diff
@@ -37,6 +39,10 @@ var (
 	}
 )
 
+func init() {
+	internal.UnregisterAuditLoggerBuilderForTesting = unregisterLoggerBuilder
```

Comment:
The intent with unregistering and re-registering was to ensure each test was fully isolated and there would be no potential side effects.
Originally instead of passing the buffer through I just had it as a global variable which is just different grossness that has to be reset every test.
Funnily enough I added the stub like this because it seemed like the accepted pattern 😆 And it is the pattern the other kind of global registries use, so that's some points towards it - keep the ick the same?

The idea of having each test register it's own version of the logger builder in the registry is interesting, but breaks with the expected pattern of how this feature should get used, so it could cause confusion to someone looking at the test code as a reference.

Thoughts? Pinging @rockspore as well to get his input here

Reply:
I also favor each tests having "isolated" registry. It's not clear to me what disadvantage there will be by having this unregister handle and I'd like to learn more about the concerns behind it.

Reply:
Why not simply use a different name for the logger in the different test cases?

IMO we shouldn't put things into `internal` like this unless there's no reasonable way to avoid it.  I don't want to end up with 1000 things in here in a few months' time.

Reply:
I just pushed 2 commits - one changes each test such that the name of the logger in the registry will be different for each test (the name is now `<test_name>_<LoggerType>`.
The second removes all occurrences of `UnregisterAuditLoggerBuilderForTesting`

Reply:
Resolving, please re-open if there is a problem